lib/charms/hydra/v0/hydra_endpoints.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Interface library for sharing hydra endpoints.
+
+This library provides a Python API for both requesting and providing public and admin endpoints.
+
+## Getting Started
+To get started using the library, you need to fetch the library using `charmcraft`.
+```shell
+cd some-charm
+charmcraft fetch-lib charms.hydra.v0.hydra_endpoints
+```
+
+To use the library from the provider side (Hydra):
```

Comment:
I don't know if this bloc is particularly useful since the only provider will ever be this hydra charm. I'd focus on the requirer side